http-client/src/main/java/io/micronaut/http/client/netty/Http1ResponseHandler.java

```diff
@@ -0,0 +1,355 @@
+                content.touch();
+                if (content.content().isReadable()) {
+                    demand -= content.content().readableBytes();
+                    streaming.add(content.content());
+                } else {
+                    content.release();
+                }
+                if (content instanceof LastHttpContent) {
+                    state = State.AFTER_CONTENT;
+                    streaming.complete();
+                    listener.finish(ctx);
+                }
+                break;
+            case DISCARDING_CONTENT:
+                ((HttpContent) msg).content().release();
+                if (msg instanceof LastHttpContent) {
+                    state = State.AFTER_CONTENT;
+                    listener.finish(ctx);
+                }
+                break;
+            case DISCARDING_CONTINUE_CONTENT:
+                ((HttpContent) msg).content().release();
+                if (msg instanceof LastHttpContent) {
+                    state = State.BEFORE_RESPONSE;
+                }
+                break;
+            case AFTER_CONTENT:
+                LOG.warn("Discarding unexpected message {}", msg);
+                ReferenceCountUtil.release(msg);
+                break;
+            default:
+                throw new AssertionError(state);
+        }
+    }
+
+    @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) throws Exception {
+        if (state == State.BUFFERED_CONTENT) {
+            devolveToStreaming(ctx);
+        }
+        if (state != State.AFTER_CONTENT && demand > 0) {
+            ctx.read();
+        } else if (state == State.BEFORE_RESPONSE || state == State.DISCARDING_CONTINUE_CONTENT || state == State.DISCARDING_CONTENT) {
+            ctx.read();
+        }
+    }
+
+    private void devolveToStreaming(ChannelHandlerContext ctx) {
+        assert state == State.BUFFERED_CONTENT;
+        assert ctx.executor().inEventLoop();
+
+        streaming = new StreamingNettyByteBody.SharedBuffer(ctx.channel().eventLoop(), listener.sizeLimits(), this);
+        if (!listener.isHeadResponse()) {
+            streaming.setExpectedLengthFrom(response.headers());
+        }
+        streamingContext = ctx;
+        if (buffered != null) {
+            for (ByteBuf buf : buffered) {
+                demand -= buf.readableBytes();
+                streaming.add(buf);
+            }
+            buffered = null;
+        }
+        state = State.UNBUFFERED_CONTENT;
+        complete(new StreamingNettyByteBody(streaming));
+    }
+
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        if (state != State.AFTER_CONTENT) {
+            exceptionCaught(ctx, new ResponseClosedException("Connection closed before response was received"));
+        }
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+        switch (state) {
```

Comment:
Can we use a new style of switch and avoid fall-through?

Reply:
ok